src/analysis/souffle/authorization_logic.dl

```diff
@@ -28,75 +28,77 @@
 
 // Some party asserts the claim that the label of an access path should
 // remove the referenced tag from the set.
-.decl saysRemoveTag(speaker: Principal, path: AccessPath, tag: Tag)
+.decl saysRemoveTag(speaker: Principal, path: AccessPath, owner: Principal, tag: Tag)
 
 // The `speaker` says that `delegatee` is allowed to indicate that a `path` does
 // not have a `tag`.
 .decl saysCanSayRemoveTag(
-  speaker: Principal, delegatee: Principal, path: AccessPath, tag: Tag)
+  speaker: Principal, delegatee: Principal, path: AccessPath, owner:Principal, tag: Tag)
 
 // Indicates that a speaker is allowed to say removeTag.
-.decl canSayRemoveTag(speaker: Principal, path: AccessPath, tag: Tag)
+.decl canSayRemoveTag(speaker: Principal, path: AccessPath, owner: Principal, tag: Tag)
 
 .decl isPrincipal(principal: Principal)
 
+// This party is the owner of a taint Tag (they have the privilege to remove
+// it).
+.decl ownsAccessPath(owner: Principal, path: AccessPath)
+
+.decl saysOwnsAccessPath(speaker: Principal, owner: Principal, path: AccessPath)
+
 // This party is the owner of a taint Tag (they have the privilege to remove
 // it).
 .decl ownsTag(owner: Principal, tag: Tag)
 
 .decl saysOwnsTag(speaker: Principal, owner: Principal, tag: Tag)
 
-// claimHasTag and saysHasTag are synonyms. They indicate that the `speaker`
-// claims that `path` definitely has `tag` taint.
-.decl claimHasTag(speaker: Principal, path: AccessPath, tag: Tag)
-.decl saysHasTag(speaker: Principal, path: AccessPath, tag: Tag)
+// The `speaker` says that `path` definitely has `tag` taint.
+.decl saysHasTag(speaker: Principal, path: AccessPath, owner: Principal, tag: Tag)
 
-// The `speaker` says that the `delegatee` can say that `path` has `tag`.
+// The `speaker` says that the `owner` can say that `path` has `tag`.
```

Comment:
It's not too hard. I think there's another minor wrinke about the translation because probably on the auth logic frontend you'd write `somePrin says canSay someFact()` (where the c and s in canSay and someFact are **lowercase**) but to get this to match with Raksha, you'd need it to say `saysCanSaySomeFact(...)` (where more letters are capitalized). Should be doable, though. (We could have a command line option for making the emitter work this way).

Reply:
#218 takes care of this.